niworkflows/utils/misc.py

```diff
@@ -313,8 +313,9 @@ def check_valid_fs_license(lic=None):
 
     if lic is not None:
         import os
-
         os.environ["FS_LICENSE"] = str(Path(lic).absolute())
```

Comment:
To me, this is still the biggest issue. Attempting to influence the environment from a helper function is not explicit, and it is likely to end up having unexpected effects. In other words: we do have a canary but we are placing it on a different coal mine.

All the environment manipulations and checks should happen in the config module. That's the only way to be sure that environment variables will have the contents we expect. I haven't checked properly, but I bet we are running this canary somewhere in the setup of fMRIPrep, very likely within a `multiprocessing.Process`. In other words, when the Process ends, this setting of `FS_LICENSE` has no effect outside the process (i.e., a different coal mine).

The canary should just call mri_convert and check the output. Just as any other nipype interface in the workflow. The resolution of the license path and setting up the environment should happen elsewhere.

Reply:
we aren't, it should be happening on the main thread prior to spawning a subprocess to build the workflow
https://github.com/poldracklab/fmriprep/blob/eb3a40b827d277ce661e31c27f6ace40ef3c2be6/fmriprep/cli/parser.py#L550-L570

When I implemented the `lic`, the idea was to try and encapsulate all licensing logic within the function. But sure, we can remove the implicit setting of the envvar. The function will just have to assume the license has been located / handled appropriately.

Reply:
The design problem is that you can't predict the result and side-effects of the function unless you are told the context in which it is run.

Reply:
Alright, so far I tested this with the follow:

- Valid license at $FREESURFER_HOME/license.txt, `True`
- Valid license at $FS_LICENSE, `True`
- No license, `False`
- Directory as license, `False`
- Invalid license, `False`